ext/reflection/php_reflection.c

```diff
@@ -2259,6 +2312,242 @@ ZEND_METHOD(reflection_parameter, __construct)
+		zend_throw_exception_ex(reflection_exception_ptr, 0,
+			"Function supplied does not have return type");
+	}
+
+	ret_reference = (return_type_reference *) emalloc(sizeof(return_type_reference));
+	ret_reference->arg_info = &fptr->common.arg_info[-1];
+	ret_reference->fptr = fptr;
+	intern->ptr = ret_reference;
+	intern->ref_type = REF_TYPE_RETURN_TYPE;
+	intern->ce = ce;
+
+	if (reference && is_closure) {
+		ZVAL_COPY_VALUE(&intern->obj, reference);
+	}
+}
+/* }}} */
+
+/* {{{ proto public static string ReflectionReturnType::export(mixed function [, bool return]) throws ReflectionException
+   */
+ZEND_METHOD(reflection_return_type, export)
+{
+	_reflection_export(INTERNAL_FUNCTION_PARAM_PASSTHRU, reflection_return_type_ptr, 1);
+}
+/* }}} */
+
+/* {{{ proto public string ReflectionReturnType::__toString(mixed function [, bool return]) throws ReflectionException
+   */
+ZEND_METHOD(reflection_return_type, __toString)
+{
+	reflection_object *intern;
+	return_type_reference *param;
+	string str;
+
+	if (zend_parse_parameters_none() == FAILURE) {
+		return;
+	}
+	GET_REFLECTION_OBJECT_PTR(param);
+	string_init(&str);
+	_parameter_string(&str, param->fptr, param->arg_info, -1, -1, "", 1);
+	RETURN_STR(str.buf);
+}
+/* }}} */
+
+/* {{{ proto public ReflectionClass|NULL ReflectionReturnType::getClass()
+   Returns this return type's class hint or NULL if there is none */
+ZEND_METHOD(reflection_return_type, getClass)
+{
+	reflection_object *intern;
+	return_type_reference *param;
+
+	if (zend_parse_parameters_none() == FAILURE) {
+		return;
+	}
+	GET_REFLECTION_OBJECT_PTR(param);
+
+	if (!param->arg_info) {
+		return;
+	}
+
+	_get_class_from_arg_info(param->fptr, param->arg_info, return_value);
+}
+/* }}} */
+
+/* {{{ proto public bool ReflectionReturnType::isArray()
+   Returns whether return type is an array */
+ZEND_METHOD(reflection_return_type, isArray)
+{
+	reflection_object *intern;
+	return_type_reference *param;
+
+	if (zend_parse_parameters_none() == FAILURE) {
+		return;
+	}
+	GET_REFLECTION_OBJECT_PTR(param);
+
+	RETVAL_BOOL(param->arg_info && param->arg_info->type_hint == IS_ARRAY);
+}
+/* }}} */
+
+/* {{{ proto public bool ReflectionReturnType::isCallable()
+   Returns whether return type is callable */
+ZEND_METHOD(reflection_return_type, isCallable)
+{
+	reflection_object *intern;
+	return_type_reference *param;
+
+	if (zend_parse_parameters_none() == FAILURE) {
+		return;
+	}
+	GET_REFLECTION_OBJECT_PTR(param);
+
+	RETVAL_BOOL(param->arg_info && param->arg_info->type_hint == IS_CALLABLE);
+}
+/* }}} */
+
+/* {{{ proto public bool ReflectionReturnType::allowsNull()
+   Returns whether NULL is allowed as this return value */
+ZEND_METHOD(reflection_return_type, allowsNull)
+{
+	reflection_object *intern;
+	return_type_reference *param;
+
+	if (zend_parse_parameters_none() == FAILURE) {
+		return;
+	}
+	GET_REFLECTION_OBJECT_PTR(param);
+
+	RETVAL_BOOL(param->arg_info->allow_null);
```

Comment:
this is the last line of this function, better use RETURN_BOOL instead of RETVAL